Libraries/LibJS/Tests/Array.prototype.reduce.js

```diff
@@ -0,0 +1,90 @@
+load("test-common.js");
+
+try {
+    assert(Array.prototype.reduce.length === 1);
+
+    assertThrowsError(() => {
+        [1].reduce();
+    }, {
+        error: TypeError,
+        message: "Array.prototype.reduce() requires at least one argument"
+    });
+
+    assertThrowsError(() => {
+        [1].reduce(undefined);
+    }, {
+        error: TypeError,
+        message: "undefined is not a function"
+    });
+
+    assertThrowsError(() => {
+        [].reduce((a, x) => x);
+    }, {
+        error: TypeError,
+        message: "Reduce of empty array with no initial value"
+
+    });
+
+    var callbackCalled = 0;
+    var callback = () => { callbackCalled++; return true };
+
+    assert([1].reduce(callback) === 1);
+    assert(callbackCalled === 0);
+
+    assert([1, 2, 3].reduce(callback) === true);
+    assert(callbackCalled === 2);
+
+    callbackCalled = 0;
+    assert([1, , , 10, , 100, , ,].reduce(callback) === true);
+    assert(callbackCalled === 2);
+
+    var constantlySad = () => ":^(";
+    var result = [].reduce(constantlySad, ":^)");
+    assert(result === ":^)");
+
+    result = [":^0"].reduce(constantlySad, ":^)");
+    assert(result === ":^(");
+
+    result = [":^0"].reduce(constantlySad);
+    assert(result === ":^0");
+
+    result = [5, 4, 3, 2, 1].reduce((accum, elem) => accum + elem);
+    assert(result === 15);
+
+    // likely parser bug: arrow func parser eats ", 100" as a part of the function, hence the parens  
```

Comment:
yep, `Parser::try_parse_arrow_function_expression()` needs to be able to take a min precedence value rather than just blindly eating up expressions (`parse_expression(0)`).

Reply:
Since this is not only about `reduce` I would leave this for another PR if that's ok.

Reply:
yes, absolutely!

Reply:
@linusg I might be wrong but it looks like it just needs to call `parse_expression(1)` instead of `0`. Is there a sense in passing this value to `Parser::try_parse_arrow_function_expression()`.

Reply:
at the moment I don't see a need for it, so we can probably just change it inside the function itself